File: MiniDOMOffice/CRM/Interazioni/SMS/SMSMessage.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases

Imports minidom.Anagrafica

Partial Public Class CustomerCalls

    ''' <summary>
    ''' Rappresenta un messaggio SMS
    ''' </summary>
    ''' <remarks></remarks>
    <Serializable> _
    Public Class SMSMessage
        Inherits CContattoUtente

        Public Sub New()
        End Sub

        Public Overrides Function GetModule() As CModule
            Return CustomerCalls.CRM.Module
        End Function

        'Public Property Numero As String
        '    Get
        '        Return Me.m_Numero
        '    End Get
        '    Set(value As String)
        '        value = Formats.ParsePhoneNumber(value)
        '        Dim oldValue As String = Me.m_Numero
        '        If (oldValue = value) Then Exit Property
        '        Me.m_Numero = value
        '        Me.DoChanged("Numero", value, oldValue)
        '    End Set
        'End Property

        Public Overrides ReadOnly Property DescrizioneAttivita As String
            Get
                Dim ret As String
                ret = "SMS " & CStr(IIf(Me.Ricevuta, "ricevuto da", "inviato a")) & " " & Me.NomePersona
                If Me.NumeroOIndirizzo <> "" Then ret &= ", tel: " & Formats.FormatPhoneNumber(Me.NumeroOIndirizzo)
                Return ret
            End Get
        End Property

        Public Overrides ReadOnly Property AzioniProposte As CCollection(Of CAzioneProposta)
            Get
                Return New CCollection(Of CAzioneProposta)
            End Get
        End Property


        'Protected Overrides Sub XMLSerialize(ByVal writer As minidom.XML.XMLWriter)
        '    MyBase.XMLSerialize(writer)
        '    writer.WriteTag("Numero", Me.m_Numero)
        'End Sub

        'Protected Overrides Sub SetFieldInternal(ByVal fieldName As String, ByVal fieldValue As Object)
        '    Select Case fieldName
        '        Case "Numero" : Me.m_Numero = XML.Utils.Serializer.DeserializeString(fieldValue)
        '        Case Else : MyBase.SetFieldInternal(fieldName, fieldValue)
        '    End Select
        'End Sub

        'Public Overrides Function GetTableName() As String
        '    Return "tbl_Telefonate"
        'End Function

        'Protected Overrides Function LoadFromRecordset(ByVal reader As DBReader) As Boolean
        '    reader.Read("Numero", Me.m_Numero)
        '    Return MyBase.LoadFromRecordset(reader)
        'End Function

        'Protected Overrides Function SaveToRecordset(ByVal writer As DBWriter) As Boolean
        '    writer.Write("Numero", Me.m_Numero)
        '    Return MyBase.SaveToRecordset(writer)
        'End Function

        Public Overrides Function GetNomeTipoOggetto() As String
            Return "SMS"
        End Function

        Public Property Numero As String
            Get
                Return Me.NumeroOIndirizzo
            End Get
            Set(value As String)
                Me.NumeroOIndirizzo = value
            End Set
        End Property

        Public Overrides Property NumeroOIndirizzo As String
            Get
                Return MyBase.NumeroOIndirizzo
            End Get
            Set(value As String)
                MyBase.NumeroOIndirizzo = Formats.ParsePhoneNumber(value)
            End Set
        End Property

    End Class



End Class

File: MiniDOMOffice/CRM/Interazioni/SMS/SMSMessage.cs
using DMD;
using DMD.Databases.Collections;
using minidom.repositories;
using System;

namespace minidom
{
    public partial class CustomerCalls
    {

        /// <summary>
        /// Rappresenta un messaggio SMS
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class SMSMessage 
            : CContattoUtente
        {

            /// <summary>
            /// Costruttore
            /// </summary>
            public SMSMessage()
            {
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            public override CContattiRepository GetContattoRepository()
            {
                return minidom.CustomerCalls.SMS;
            }

            /// <summary>
            /// Descrizione attività
            /// </summary>
            public override string DescrizioneAttivita
            {
                get
                {
                    string ret;
                    ret = "SMS " + (Ricevuta? "ricevuto da" : "inviato a") + " " + NomePersona;
                    if (!string.IsNullOrEmpty(NumeroOIndirizzo))
                        ret += ", tel: " + Sistema.Formats.FormatPhoneNumber(NumeroOIndirizzo);
                    return ret;
                }
            }

            /// <summary>
            /// Azioni proposte
            /// </summary>
            public override CCollection<CAzioneProposta> AzioniProposte
            {
                get
                {
                    return new CCollection<CAzioneProposta>();
                }
            }

             
            /// <summary>
            /// Tipo oggetto
            /// </summary>
            /// <returns></returns>
            public override string GetNomeTipoOggetto()
            {
                return "SMS";
            }

            /// <summary>
            /// Numero della controparte
            /// </summary>
            public string Numero
            {
                get
                {
                    return NumeroOIndirizzo;
                }

                set
                {
                    NumeroOIndirizzo = value;
                }
            }

            /// <summary>
            /// Numero della controparte
            /// </summary>
            public override string NumeroOIndirizzo
            {
                get
                {
                    return base.NumeroOIndirizzo;
                }

                set
                {
                    base.NumeroOIndirizzo = Sistema.Formats.ParsePhoneNumber(value);
                }
            }

            
        }
    }
}